airbyte-cdk/bulk/toolkits/load-gcs/src/test/kotlin/io/airbyte/cdk/load/file/gcs/GcsClientTest.kt

```diff
@@ -0,0 +1,291 @@
+        return blob
+    }
+
+    @BeforeEach
+    fun setup() {
+        storage = mockk()
+        val auth = GcsHmacKeyConfiguration("test-access-key", "test-secret-key")
+        config = GcsClientConfiguration(bucketName, bucketPath, auth, region)
+        gcsClient = GcsClient(storage, config)
+    }
+
+    @Test
+    fun `test list with empty prefix`() = runBlocking {
+        val blob1 = mockBlob("test-file1")
+        val blob2 = mockBlob("test-file2")
+        val blobs = mockk<Iterable<Blob>>()
+
+        every { storage.list(bucketName, any()) } returns
+            mockk { every { iterateAll() } returns blobs }
+        every { blobs.iterator() } returns listOf(blob1, blob2).iterator()
+
+        val results = gcsClient.list("").toList()
+
+        assertEquals(2, results.size)
+        assertEquals("test-file1", results[0].key)
+        assertEquals("test-file2", results[1].key)
+        assertEquals(config, results[0].storageConfig)
+        assertEquals(config, results[1].storageConfig)
+
+        verify { storage.list(bucketName, any()) }
+    }
+
+    @Test
+    fun `test list with prefix`() = runBlocking {
+        val prefix = "subfolder"
+        val fullPrefix = "$bucketPath/$prefix"
+        val blob1 = mockBlob("$fullPrefix/test-file1")
+        val blobs = mockk<Iterable<Blob>>()
+
+        every { storage.list(bucketName, any()) } returns
+            mockk { every { iterateAll() } returns blobs }
+        every { blobs.iterator() } returns listOf(blob1).iterator()
+
+        val results = gcsClient.list(prefix).toList()
+
+        assertEquals(1, results.size)
+        assertEquals("$fullPrefix/test-file1", results[0].key)
+
+        verify { storage.list(bucketName, any()) }
+    }
+
+    @Test
+    fun `test move`() = runBlocking {
+        val sourceKey = "source-file"
+        val targetKey = "target-file"
+        val fullSourceKey = "$bucketPath/$sourceKey"
+        val fullTargetKey = "$bucketPath/$targetKey"
+
+        val sourceBlobId = BlobId.of(bucketName, fullSourceKey)
+        val targetBlobId = BlobId.of(bucketName, fullTargetKey)
+        val copyRequestSlot = slot<Storage.CopyRequest>()
+        val copyWriter = mockk<CopyWriter>()
+
+        every { storage.copy(capture(copyRequestSlot)) } returns copyWriter
+        every { storage.delete(sourceBlobId) } returns true
+
+        val result = gcsClient.move(sourceKey, targetKey)
+
+        assertEquals(targetKey, result.key)
+        assertEquals(config, result.storageConfig)
+
+        assertEquals(sourceBlobId.bucket, copyRequestSlot.captured.source.bucket)
+        assertEquals(sourceBlobId.name, copyRequestSlot.captured.source.name)
+        assertEquals(targetBlobId.bucket, copyRequestSlot.captured.target.bucket)
+        assertEquals(targetBlobId.name, copyRequestSlot.captured.target.name)
+
+        verify { storage.delete(sourceBlobId) }
+    }
+
+    @Test
+    fun `test move with remoteObject`() = runBlocking {
```

Comment:
this test seems reasonable, but I feel like we should just add a default implementation for `move(remoteObject, toKey)` onto the interface >.>